vacuum_world.ipynb

```diff
@@ -336,21 +474,35 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "To create our agent, we need two functions: INTERPRET-INPUT function, which generates an abstracted description of the current state from the percerpt and the RULE-MATCH function, which returns the first rule in the set of rules that matches the given state description."
+    "To create our agent, we need two functions: INTERPRET-INPUT function, which generates an abstracted description of the current state from the percerpt and the RULE-MATCH function, which returns the first rule in the set of rules that matches the given state description. For rules, we will implement a `Rule` class."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 13,
+   "execution_count": 17,
    "metadata": {},
    "outputs": [],
    "source": [
     "# TODO: Implement these functions for two-dimensional environment\n",
     "# Interpret-input function for the two-state environment\n",
-    "def interpret_input(percept):\n",
-    "    pass\n",
+    "# Rule class\n",
+    "class Rule:\n",
+    "        def __init__(self, state, action):\n",
+    "            self.__state = state\n",
+    "            self.action = action\n",
+    "            \n",
+    "        def matches(self, state):\n",
+    "            return self.__state == state\n",
+    "        \n",
+    "def interpret_input(state):\n",
+    "    return state\n",
+    "    \n",
+    "loc_A = (0, 0)\n",
+    "loc_B = (1, 0)\n",
     "\n",
-    "rules = None\n",
+    "# create rules for a two state Vacuum Environment\n",
```

Comment:
Add a hyphen ( '-' ) between 'two' and 'state'.

Reply:
Fixed the typo. Review it whenever you are free @MrDupin